src/gluonts/representation/embedding.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from .representation import Representation
+from .custom_binning import CustomBinning
+from .global_relative_binning import GlobalRelativeBinning
+from .local_absolute_binning import LocalAbsoluteBinning
+
+# Standard library imports
+from typing import Tuple, Optional, Union, List
+
+from mxnet.gluon import nn
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.model.common import Tensor
+from gluonts.dataset.common import Dataset
+
+
+Binning = Union[CustomBinning, GlobalRelativeBinning, LocalAbsoluteBinning]
+
+
+class Embedding(Representation):
+    """
+    A class representing an embedding operation on top of a given binning.
+    
+    Parameters
+    ----------
+    binning
+        The underlying binning.
+    size
+        The desired embedding size.
+        (default: round(num_bins**(1/4)))
+    """
+
+    @validated()
+    def __init__(self, binning: Binning, size: int = -1, *args, **kwargs):
```

Comment:
For optional arguments (size in this case) it’s better to use Optional (from typing import Optional) and default to None

Reply:
Oh yeah right, that's indeed way cleaner!